assets/training/model_management/src/azureml/model/mgmt/processors/pyfunc/vision/detection_predict.py

```diff
@@ -171,26 +172,26 @@ def predict(self, context: mlflow.pyfunc.PythonModelContext,
         :return: Output of inferencing
         :rtype: Pandas DataFrame with columns ["boxes"] for object detection
         """
-        # process the images in image column
-        processed_images = input_data.loc[:, [MLflowSchemaLiterals.INPUT_COLUMN_IMAGE]].apply(
-            axis=1, func=process_image_pandas_series
-        )
+        # Read all the input images.
+        np_images = [
+            np.array(Image.open(io.BytesIO(process_image(image))))
+            for image in input_data[MLflowSchemaLiterals.INPUT_COLUMN_IMAGE]
+        ]
+
         if not params:
             params = {}
         text_prompt = params.get(MMDetLiterals.TEXT_PROMPT, None)
         custom_entities = params.get(MMDetLiterals.CUSTOM_ENTITIES, True)
         if not text_prompt and self.language_model:
             raise ValueError("text prompt not sent. please send text_prompt for Launguage models")
         classes = text_prompt.split(". ") if self.language_model else self.classes
-        with tempfile.TemporaryDirectory() as tmp_output_dir:
-            image_path_list = (
-                processed_images.iloc[:, 0].map(lambda row: create_temp_file(row, tmp_output_dir)[0]).tolist()
-            )
```

Comment:
Should we change the same logic in FT'ed model predict script as well(for HF and MMD)
https://msdata.visualstudio.com/Vienna/_git/AzureMlCli?path=/src/azureml-acft-image-components/azureml/acft/image/components/finetune/common/mlflow/mmdet_mlflow_model_wrapper.py&version=GBnvijayrania/add-diffusion-data-class&line=162&lineEnd=167&lineStartColumn=1&lineEndColumn=13&lineStyle=plain&_a=contents
cc: @rjaincc

Reply:
I would say yes, but since that's a different repo, someone else should take it. Made ticket for Rupal - please reassign.

Reply:
@rdondera-microsoft - Let's please take care of this as part of the same work item. It would be very easy for the base and FT-ed model to go out of sync - we run the risk of seeing unexpected issues.